minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/ConfigSchema.java

```diff
@@ -96,51 +100,154 @@ public ConfigSchema(Map map) {
+
+        return processors;
     }
 
-    private void checkForDuplicateNames(String errorMessagePrefix, List<String> names) {
-        if (names != null) {
-            Set<String> seenNames = new HashSet<>();
-            Set<String> duplicateNames = new TreeSet<>();
-            for (String name : names) {
-                if (!seenNames.add(name)) {
-                    duplicateNames.add(name);
+    protected List<ConnectionSchema> getConnectionSchemas(List<Map> connectionMaps) {
+        if (connectionMaps == null) {
+            return null;
+        }
+        List<ConnectionSchema> connections = convertListToType(connectionMaps, "connection", ConnectionSchema.class, CONNECTIONS_KEY);
+        Map<String, Integer> idMap = connections.stream().map(ConnectionSchema::getId).filter(
+                s -> !StringUtil.isNullOrEmpty(s)).collect(Collectors.toMap(Function.identity(), s -> 2, Integer::compareTo));
+
+        Map<String, String> processorNameToIdMap = new HashMap<>();
+
+        // We can't look up id by name for names that appear more than once
+        Set<String> duplicateProcessorNames = new HashSet<>();
+
+        List<ProcessorSchema> processors = getProcessors();
+        if (processors != null) {
+            processors.stream().forEachOrdered(p -> processorNameToIdMap.put(p.getName(), p.getId()));
+
+            Set<String> processorNames = new HashSet<>();
+            processors.stream().map(ProcessorSchema::getName).forEachOrdered(n -> {
+                if (!processorNames.add(n)) {
+                    duplicateProcessorNames.add(n);
                 }
+            });
+        }
+
+        Set<String> remoteInputPortIds = new HashSet<>();
+        List<RemoteProcessingGroupSchema> remoteProcessingGroups = getRemoteProcessingGroups();
+        if (remoteProcessingGroups != null) {
+            remoteInputPortIds.addAll(remoteProcessingGroups.stream().filter(r -> r.getInputPorts() != null)
+                    .flatMap(r -> r.getInputPorts().stream()).map(RemoteInputPortSchema::getId).collect(Collectors.toSet()));
+        }
+
+        Set<String> problematicDuplicateNames = new HashSet<>();
+        // Set unset ids
+        connections.stream().filter(connection -> StringUtil.isNullOrEmpty(connection.getId())).forEachOrdered(connection -> connection.setId(getUniqueId(idMap, connection.getName())));
+
+        connections.stream().filter(connection -> StringUtil.isNullOrEmpty(connection.getSourceId())).forEach(connection -> {
+            String sourceName = connection.getSourceName();
+            if (remoteInputPortIds.contains(sourceName)) {
+                connection.setSourceId(sourceName);
+            } else {
+                if (duplicateProcessorNames.contains(sourceName)) {
+                    problematicDuplicateNames.add(sourceName);
+                }
+                connection.setSourceId(processorNameToIdMap.get(sourceName));
```

Comment:
This (and the corresponding line for dest names) should check if source name is actually in the list of processors (it's already checking if it's in the list of remote input ports). I tested a flow where I didn't put the right name for "source name" and got this confusing error message: 

Exception in thread "main" java.io.IOException: Unable to successfully transform the provided configuration
	at org.apache.nifi.minifi.bootstrap.RunMiNiFi.performTransformation(RunMiNiFi.java:1543)
	at org.apache.nifi.minifi.bootstrap.RunMiNiFi.start(RunMiNiFi.java:1083)
	at org.apache.nifi.minifi.bootstrap.RunMiNiFi.main(RunMiNiFi.java:215)
Caused by: org.apache.nifi.minifi.bootstrap.exception.InvalidConfigurationException: Failed to transform config file due to:['source id' in section 'Connections' because it was not found and it is required]
	at org.apache.nifi.minifi.bootstrap.util.ConfigTransformer.transformConfigFile(ConfigTransformer.java:92)
	at org.apache.nifi.minifi.bootstrap.RunMiNiFi.performTransformation(RunMiNiFi.java:1539)
	... 2 more